src/main/software/amazon/event/ruler/input/StringValueParser.java

```diff
@@ -0,0 +1,10 @@
+package software.amazon.event.ruler.input;
+
+public interface StringValueParser {
```

Comment:
I think I need one more input on the inclusion of these parsers. @jonessha  do you have an opinions on either keeping / removing / changing interfaces?

Reply:
I'm not strongly opposed to this change, but at the same time, I don't really get the purpose of including all these interfaces. In my mind for how Ruler will evolve, these interfaces will serve no purpose. @h1alexbel - curious what benefit you see these interfaces providing as the code base evolves?

Reply:
> I'm not strongly opposed to this change, but at the same time, I don't really get the purpose of including all these interfaces. In my mind for how Ruler will evolve, these interfaces will serve no purpose. @h1alexbel - curious what benefit you see these interfaces providing as the code base evolves?

Benefit is maintainability in the future and possible API extension via interfaces, and that all classes will be used by some special contract.

Reply:
looks like this PR has stalled. since there's no blockers with any changes Shawn has staged in a different branch / repo; I don't mind merging this for now. We can always undo if we think this creates too much friction in future.